src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -220,12 +218,9 @@ export class KernelExecution implements IDisposable {
 
     private async executeIndividualCell(
         kernelPromise: Promise<IKernel>,
-        cellExecution: CellExecution
+        cellExecution: CellExecution,
+        notebook: INotebook
     ): Promise<NotebookCellRunState | undefined> {
-        if (!this.notebook) {
```

Comment:
We no longer need this hack, we now pass `notebook` as an argument. Cleaner, else calling code needs to ensure property is set before invoking this class methods (not inutitive & poor design)